### PR TITLE
[cppgraphqlgen] Update to v3.3.0

### PR DESCRIPTION
**Pull request by wravery**
**Describe the pull request**

- What does your PR fix?

Updates to the v3.3.0 release of https://github.com/microsoft/cppgraphqlgen. This has been tested along with #14841 on Windows, Mac, and Linux, but it works with the prior version of PEGTL as well.

- Which triplets are supported/not supported? Have you updated the CI baseline?

This version has the same `std::filesystem` auto-detection for `schemagen` as PEGTL in #14841. It should work on any target triplet and with all of the runners in the CI workflow.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

**Comment by JackBoosY** [comment]
See my comment https://github.com/microsoft/vcpkg/pull/14841#issuecomment-736186228

File: ports/cppgraphqlgen/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: cppgraphqlgen
-Version: 3.2.4
-Build-Depends: boost-filesystem (!uwp&!windows), boost-program-options, pegtl, rapidjson
+Version: 3.3.0
+Build-Depends: boost-program-options, pegtl, rapidjson
 Description: C++ GraphQL schema service generator
 Homepage: https://github.com/microsoft/cppgraphqlgen
```
Comment (JonLiu1993) [inline ports/cppgraphqlgen/CONTROL:5]: It is recommended to add the linux not supported in the baseline here, for example: Supports: !linux

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by wravery** [comment]
This depends on #14841 now.

**Review by ras0219 — APPROVED** [review]

File: ports/cppgraphqlgen/CONTROL [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Source: cppgraphqlgen
-Version: 3.2.4
-Build-Depends: boost-filesystem (!uwp&!windows), boost-program-options, pegtl, rapidjson
+Version: 3.3.0
```
Comment (JackBoosY) [inline ports/cppgraphqlgen/CONTROL:2]:
```suggestion
Version: 2020-12-07
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**